src/DotLiquid/Strainer.cs

```diff
@@ -31,16 +34,26 @@ public static void GlobalFilter(string rawName, object target, MethodInfo method
             FilterFuncs[name] = Tuple.Create(target, methodInfo);
         }
 
+        /// <summary>
+        /// Creates a new instance of the <see cref="Strainer"/> class.
+        /// </summary>
+        /// <param name="context">The DotLiquid context to be used for creating the <see cref="Strainer"/>.</param>
+        /// <returns>A new instance of the <see cref="Strainer"/> class.</returns>
         public static Strainer Create(Context context)
         {
-            Strainer strainer = new Strainer(context);
+            var strainer = new Strainer(context);
+
+            if (context.SyntaxCompatibilityLevel < SyntaxCompatibility.DotLiquidLatest)
+                strainer.Extend(context.SyntaxCompatibilityLevel, typeof(LegacyFilters));
+
+            strainer.Extend(context.SyntaxCompatibilityLevel, typeof(StandardFilters));
 
             foreach (var keyValue in Filters)
-                strainer.Extend(keyValue.Value);
+                strainer.Extend(context.SyntaxCompatibilityLevel, keyValue.Value);
 
             foreach (var keyValue in FilterFuncs)
                 strainer.AddMethodInfo(keyValue.Key, keyValue.Value.Item1, keyValue.Value.Item2);

```

Comment:
I know this isn't changed, but using a Tuple instead of a custom class/struct/record makes this fairly unreadable. I had to go to the definition, then check where is was being populated to understand what's going on.